### PR TITLE
Prevent code execution in Directives ctor's

### LINKED ISSUE #133: Code execution in Directives() ctor
### Expected Behavior

No code execution in any ctor of the `Directives()` class

### Actual Behavior

Code execution in the following ctors:
- line 94:
```csharp
    public Directives(IText text) : this(text.AsString())
    { }
```
- line 101:
```csharp
    public Directives(String text) : this(Directives.Parse(text))
    { }
```
- line 115:
```csharp
    public Directives(IEnumerable<IDirective> dirs)
    {
        this.Append(dirs);
    }
```

### PR DESCRIPTION
**Pull request by aveXcaesar**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

close #133 

### What is the new behavior?

- No code execution in `Directives()` ctor's

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: src/Yaapii.Xambly/Directive/Directives.cs [hunk 1]
```diff
@@ -74,47 +72,92 @@
...
 
     /// <summary>
     /// ctor
     /// </summary>
     /// <param name="dirs">directives</param>
-    public Directives(IEnumerable<IDirective> dirs)
+    public Directives(IEnumerable<IDirective> dirs) : this(
+        new StickyScalar<ICollection<IDirective>>(() =>
+            new ThreadsafeCollection<IDirective>(
+                new Object(),
```
[the remark this answers is not in the document]
Comment (aveXcaesar) [reply src/Yaapii.Xambly/Directive/Directives.cs]: Object() is the sync-object for the ThreadsafeCollection.
I added a new ctor and found code execution in the ThreadsaveCollection class. -> fixed

**Comment by aveXcaesar** [comment]
@Meerownymous 
changes done

File: src/Yaapii.Xambly/Directive/Directives.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aveXcaesar) [reply src/Yaapii.Xambly/Directive/Directives.cs]: Update: Remove refactoring in class ThreadSaveColletion. This is part of issue #136!

File: src/Yaapii.Xambly/Directive/Directives.cs [hunk 2]
```diff
@@ -64,57 +69,100 @@ namespace Yaapii.Xambly
...
 
         /// <summary>
-        /// ctor.
+        /// Collection of <see cref="IDirective"/>.
         /// </summary>
-        /// <param name="text">Xambly script</param>
-        public Directives(String text) : this(Directives.Parse(text))
+        /// <param name="xambly">Xambly script</param>
+        public Directives(IText xambly) : this(
+            new StickyScalar<ICollection<IDirective>>(() =>
```
Comment (Meerownymous) [inline src/Yaapii.Xambly/Directive/Directives.cs:110]: should be a Solid to ensure threadsafeness.

File: src/Yaapii.Xambly/ThreadSafeCollection.cs [hunk 3]
```diff
@@ -43,6 +42,18 @@ public ThreadsafeCollection(object syncRoot)
             this.sync = syncRoot;
         }
 
+        /// <summary>
+        /// A collection which is threadsafe.
+        /// </summary>
+        /// <param name="items">list to instantiate from</param>
+        public ThreadsafeCollection(IEnumerable<T> items)
+        {
+            if (items == null)
```
Comment (Meerownymous) [inline src/Yaapii.Xambly/ThreadSafeCollection.cs:51]: code execution in ctor

Comment (Meerownymous) [inline src/Yaapii.Xambly/ThreadSafeCollection.cs:51]: ctor should be placed in a lock statement

**Comment by Meerownymous** [comment]
Closed due to inactivity.

**Pull request closed without merging**